File: chapter_1/simple_shift_cipher/simple_shift_cipher.cpp
#include <bits/stdc++.h>

using namespace std;

void simpleShiftCipher(string &str, int shift){
    for(int i=0; i<str.size(); i++){
        if(isupper(str[i]))
        str[i] = char((str[i]+shift-65)%26+65);
        else
        str[i] = char((str[i]+shift-97)%26+97);
    }
    return;
}

int main(){
    // cout << int('z');
    string str; cin>>str;
    int shift; cin>>shift;
    simpleShiftCipher(str, shift);
    cout << str;
    
    return 0;
}

File: chapter_1/simple_shift_cipher/simple_shift_cipher.py

def simple_shift_cipher(s, sentence):
    words = sentence.split(' ')
    i = 0
    for text in words:
        result = ""
        for i in range(len(text)):
            char = text[i]
            if (char.isupper()):
                result += chr((ord(char) + s-65) % 26 + 65)
            else:
                result += chr((ord(char) + s - 97) % 26 + 97)
        words[i] = result
        i+=1
    sentence = ' '.join(words)
    return sentence


if __name__ == '__main__':
    shift = int(input("Enter the shift: "))
    sentence = input("Enter the sentence: ")
    sentence = simple_shift_cipher(s=shift, sentence=sentence)
    print(f"ciphered: {sentence}\n")
